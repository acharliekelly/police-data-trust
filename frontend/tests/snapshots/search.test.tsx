import * as React from "react"
import Dashboard from "../../pages/search"
import { render, router, setAuthForTest, userEvent, waitFor } from "../test-utils"

beforeAll(() => setAuthForTest())

it("renders Dashboard correctly", async () => {
  // Ensure consistent title id's for fontawesome
  jest.spyOn(global.Math, "random").mockImplementation(() => 0)
  const { container } = render(<Dashboard />)
  expect(container).toMatchSnapshot()
  jest.clearAllMocks
})

it("redirects to login on logout", async () => {
  const { getByRole } = render(<Dashboard />)
  const logout = getByRole("button", { name: /logout/i })

  userEvent.click(logout)

  await waitFor(() => expect(router.pathname).toBe("/login"))
})
