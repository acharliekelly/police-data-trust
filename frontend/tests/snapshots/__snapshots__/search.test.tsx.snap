// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard correctly 1`] = `
<div>
  <div>
    <header
      class="wrapper"
    >
      <div
        class="backgroundBanner"
      >
        <div
          class="leftHeader"
        >
          <img
            alt="A blue circular logo design reads 'National Police Data Coalition'"
            class="logo medium"
            src="/NPDCLogo.svg"
          />
          <div
            class="titleContainer"
          >
            <h2>
              National Police Data Coalition
            </h2>
            <p>
              The national index of police incidents
            </p>
          </div>
        </div>
        <nav
          aria-label="Main Navigation"
        >
          <div
            class="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              role="button"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-ellipsis-v fa-w-6 fa-3x "
                data-icon="ellipsis-v"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 192 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="hidden"
              role="menu"
            >
              <ul
                id="navMenu"
              >
                <li
                  class=""
                >
                  <a
                    href="/search"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Search
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/profile"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Profile
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/logout"
                    role="menuitem"
                    tabindex="-1"
                  >
                    Sign Out
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/"
                    role="menuitem"
                    tabindex="-1"
                  >
                    DONATE
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <ul
            id="navMenu"
          >
            <li
              class=""
            >
              <a
                href="/search"
                role="menuitem"
                tabindex="-1"
              >
                Search
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/profile"
                role="menuitem"
                tabindex="-1"
              >
                Profile
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/logout"
                role="menuitem"
                tabindex="-1"
              >
                Sign Out
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/"
                role="menuitem"
                tabindex="-1"
              >
                DONATE
              </a>
            </li>
          </ul>
          <button
            class="primaryButton"
            type="submit"
          >
            DONATE
          </button>
          <button
            class="primaryButton"
            type="submit"
          >
            LOGOUT
          </button>
        </nav>
      </div>
    </header>
    <div
      style="display: flex; flex-direction: row; justify-content: center;"
    >
      <button
        class="primaryButton"
        style="cursor: pointer; margin: 2em 1em 1em 1em;"
        type="button"
      >
        bubble
      </button>
      <button
        class="primaryButton"
        style="cursor: pointer; margin: 2em 1em 1em 1em;"
        type="button"
      >
        map
      </button>
    </div>
    <div
      class="mapContainer"
      id="map-container"
    >
      <div
        class="mapWrapper"
      >
        <svg
          class="mapData"
          id="map-svg"
          viewBox="0, 0, 1200, 700"
        >
          <g
            id="zoom-container"
          >
            <svg
              id="map"
              viewBox="0, 0, 1200, 700"
            />
            <svg
              class="markerLayer"
              height="100%"
              viewBox="0, 0, 1200, 700"
              width="100%"
            />
          </g>
        </svg>
      </div>
      <div
        class="mapButton"
      >
        <button
          class="primaryButton"
          id="zoom-reset"
        >
          Reset Zoom
        </button>
      </div>
      <div
        style="position: absolute; right: 0px; bottom: 0px;"
      >
        <div
          class="chartKeyContainer"
          id="map-key"
        >
          <div
            class="KeyEntry"
          >
            <div
              class="keySymbolEntry"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="var(--darkBlue)"
                  r="10"
                />
              </svg>
              <label
                class="label"
                for="Data Rich Cities"
              >
                Data Rich Cities
              </label>
            </div>
          </div>
          <div
            class="KeyEntry"
          >
            <div
              class="keySymbolEntry"
            >
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="var(--lightBlue)"
                  height="20"
                  width="20"
                />
              </svg>
              <label
                class="label"
                for="Data Poor Cities"
              >
                Data Poor Cities
              </label>
            </div>
          </div>
          <div
            class="KeyEntry"
          >
            <div />
            <div
              class="keyGradientEntry"
            >
              <label
                class="label"
              >
                Less Data
              </label>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(124, 174, 215)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(105, 144, 186)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(86, 113, 157)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(67, 83, 128)"
                  height="20"
                  width="20"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <rect
                  fill="rgb(48, 52, 99)"
                  height="20"
                  width="20"
                />
              </svg>
              <label
                class="label"
              >
                More Data
              </label>
            </div>
          </div>
          <div
            class="KeyEntry"
          >
            <div />
            <div
              class="keyGradientEntry"
            >
              <label
                class="label"
              >
                Blue
              </label>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(124, 174, 215)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(119, 156, 187)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(113, 138, 159)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(108, 120, 130)"
                  r="10"
                />
              </svg>
              <svg
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <circle
                  cx="10"
                  cy="10"
                  fill="rgb(102, 102, 102)"
                  r="10"
                />
              </svg>
              <label
                class="label"
              >
                Grey
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <table
      aria-label="Search Results"
      class="dataTable"
      role="table"
    >
      <thead
        class="dataHeader"
      >
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Officer(s) Involved
            <span
              class="sortArrow"
            >
                
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <span
              class="columnHead"
            >
              Date/Time
              <div
                aria-describedby="dateTimeTooltip"
                aria-label="Learn more about Date Time"
                class="tooltipContainer"
                role="tooltip"
                tabindex="0"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-info-circle fa-w-16 fa-xs tooltipIcon"
                  data-icon="info-circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                    fill="currentColor"
                  />
                </svg>
                <div
                  class="tooltip"
                  id="dateTimeTooltip"
                >
                  <p>
                    Dates are presented in Month-Day-Year format
                  </p>
                  <p>
                    Times are written in 24-hour format
                  </p>
                  <p>
                    Any Tildes (~) indicate approximations
                  </p>
                </div>
              </div>
            </span>
            <span
              class="sortArrow"
            >
                
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Incident Type
            <span
              class="sortArrow"
            >
                
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <span
              class="columnHead"
            >
              Use of Force
              <div
                aria-describedby="useOfForceTooltip"
                aria-label="Learn more about Use Of Force"
                class="tooltipContainer"
                role="tooltip"
                tabindex="0"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-info-circle fa-w-16 fa-xs tooltipIcon"
                  data-icon="info-circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                    fill="currentColor"
                  />
                </svg>
                <div
                  class="tooltip"
                  id="useOfForceTooltip"
                >
                  <p>
                    Use of Force refers to the continuum of violence that may be used in an incident
                  </p>
                  <p>
                    For more info, view our Use of Force explainer
                  </p>
                </div>
              </div>
            </span>
            <span
              class="sortArrow"
            >
                
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            Source
            <span
              class="sortArrow"
            >
                
            </span>
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            <span
              class="columnHeadIcon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sliders-h fa-w-16 fa-lg "
                data-icon="sliders-h"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="sortArrow"
            >
                
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            B.Eilish
          </td>
          <td
            role="cell"
          >
            2/24/2020
          </td>
          <td
            role="cell"
          >
            Record Of The Year
          </td>
          <td
            role="cell"
          >
            Verbalization
          </td>
          <td
            role="cell"
          >
            Bad Guy
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            B.Iver
          </td>
          <td
            role="cell"
          >
            1/28/2021
          </td>
          <td
            role="cell"
          >
            Record Of The Year
          </td>
          <td
            role="cell"
          >
            Armed presence
          </td>
          <td
            role="cell"
          >
            Hey, Ma
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            A.Grande
          </td>
          <td
            role="cell"
          >
            3/1/2020
          </td>
          <td
            role="cell"
          >
            Record Of The Year
          </td>
          <td
            role="cell"
          >
            Injurious restraint
          </td>
          <td
            role="cell"
          >
            7 rings
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            Khalid
          </td>
          <td
            role="cell"
          >
            4/1/2020
          </td>
          <td
            role="cell"
          >
            Record Of The Year
          </td>
          <td
            role="cell"
          >
            Verbalization
          </td>
          <td
            role="cell"
          >
            Talk
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            L.NasX, B.Cyrus
          </td>
          <td
            role="cell"
          >
            2/6/2020
          </td>
          <td
            role="cell"
          >
            Record Of The Year
          </td>
          <td
            role="cell"
          >
            Verbalization
          </td>
          <td
            role="cell"
          >
            Old Town Road
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            Lizzo
          </td>
          <td
            role="cell"
          >
            3/7/2021
          </td>
          <td
            role="cell"
          >
            Record Of The Year
          </td>
          <td
            role="cell"
          >
            Unarmed presence
          </td>
          <td
            role="cell"
          >
            Truth Hurts
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            P.Malone, S.Lee
          </td>
          <td
            role="cell"
          >
            1/22/2021
          </td>
          <td
            role="cell"
          >
            Record Of The Year
          </td>
          <td
            role="cell"
          >
            Unarmed blunt impact
          </td>
          <td
            role="cell"
          >
            Sunflower
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            B.Eilish
          </td>
          <td
            role="cell"
          >
            6/9/2020
          </td>
          <td
            role="cell"
          >
            Album Of The Year
          </td>
          <td
            role="cell"
          >
            Armed presence
          </td>
          <td
            role="cell"
          >
            When We All Fall Asleep...
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            B.Iver
          </td>
          <td
            role="cell"
          >
            12/12/2019
          </td>
          <td
            role="cell"
          >
            Album Of The Year
          </td>
          <td
            role="cell"
          >
            Armed presence
          </td>
          <td
            role="cell"
          >
            I,I
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="dataRows"
          role="row"
        >
          <td
            role="cell"
          >
            A.Grande
          </td>
          <td
            role="cell"
          >
            8/12/2020
          </td>
          <td
            role="cell"
          >
            Album Of The Year
          </td>
          <td
            role="cell"
          >
            Unarmed blunt impact
          </td>
          <td
            role="cell"
          >
            thank u, next
          </td>
          <td>
            <svg
              aria-labelledby="svg-inline--fa-title-000000000000"
              class="svg-inline--fa fa-plus-circle fa-w-16 actionBtn"
              data-icon="plus-circle"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-000000000000"
              >
                Save Record
              </title>
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                fill="currentColor"
              />
            </svg>
          </td>
        </tr>
      </tbody>
      <tfoot
        class="dataFooter"
      >
        <tr>
          <td
            class="recordCount"
          >
            <strong>
              163
               records found
            </strong>
          </td>
          <td
            colspan="2"
          />
          <td
            colspan="2"
          >
            <button
              class="pageBtn"
              disabled=""
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-angle-left fa-w-8 "
                data-icon="angle-left"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 256 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <span
              class="pageCnt"
            >
              1
               of 
              17
            </span>
            <button
              class="pageBtn"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-angle-right fa-w-8 "
                data-icon="angle-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 256 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </td>
          <td />
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
